jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsHelper.kt

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.nodejs
+
+import com.intellij.lang.javascript.buildTools.npm.PackageJsonUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectFileIndex
+import com.intellij.openapi.vfs.VirtualFile
+
+/**
+ * WebStorm doesn't allow user to mark a folder as source root. This method infers a folder as source root based on
```

Comment:
What is this inference used for?

Reply:
This is used for detecting the source root for the Node project. Unlike IntelliJ or PyCharm, in WebStorm, you cannot mark a folder as source root. Thus, we cannot simply use 
`ProjectFileIndex.getInstance(project).getSourceRootForFile(virtualFile)` as the source root.

This method is used for generating the fully qualified handler name.